File: app_manager/appmanager/src/main/java/app/editors/manager/ui/fragments/login/SplashFragment.java
package app.editors.manager.ui.fragments.login;

import android.content.res.Resources;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.view.InflateException;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatImageView;
import androidx.core.content.ContextCompat;

import java.util.concurrent.TimeUnit;

import app.editors.manager.R;
import app.editors.manager.managers.utils.FirebaseUtils;
import app.editors.manager.ui.activities.main.MainActivity;
import app.editors.manager.ui.fragments.base.BaseAppFragment;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;
import io.reactivex.schedulers.Schedulers;

public class SplashFragment extends BaseAppFragment {

    public static final String TAG = SplashFragment.class.getSimpleName();

    private static final int DELAY_SPLASH = 1000;

    @BindView(R.id.appCompatImageView)
    protected AppCompatImageView mAppCompatImageView;

    private Unbinder mUnbinder;
    private Disposable mDisposable;

    public static SplashFragment newInstance() {
        return new SplashFragment();
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        final View view = inflater.inflate(R.layout.fragment_splash, container, false);
        mUnbinder = ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        init();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        mUnbinder.unbind();
        if (mDisposable != null) {
            mDisposable.dispose();
        }
    }

    private void init() {
        try {
            Drawable drawable = ContextCompat.getDrawable(requireContext(), R.drawable.image_onlyoffice_text);
            mAppCompatImageView.setImageDrawable(drawable);
        } catch (InflateException | Resources.NotFoundException e) {
            FirebaseUtils.addCrash("Inflate error when start");
            e.printStackTrace();
        }

        mDisposable = Observable.just(1).delay(DELAY_SPLASH, TimeUnit.MILLISECONDS)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(integer -> {
                    MainActivity.show(getContext());
                    requireActivity().finish();
                });
    }

}

File: app_manager/appmanager/src/main/java/app/editors/manager/ui/fragments/login/SplashFragment.kt
package app.editors.manager.ui.fragments.login

import android.content.Context
import android.content.res.Resources
import android.os.Bundle
import android.view.InflateException
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import app.editors.manager.app.appComponent
import app.editors.manager.databinding.FragmentSplashBinding
import app.editors.manager.managers.tools.PreferenceTool
import app.editors.manager.managers.utils.FirebaseUtils
import app.editors.manager.ui.activities.main.MainActivity
import app.editors.manager.ui.activities.main.PasscodeActivity
import app.editors.manager.ui.fragments.base.BaseAppFragment
import io.reactivex.Observable
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import io.reactivex.schedulers.Schedulers
import java.util.concurrent.TimeUnit
import javax.inject.Inject

class SplashFragment : BaseAppFragment() {


    companion object {
        val TAG: String = SplashFragment::class.java.simpleName

        private const val DELAY_SPLASH = 1000

        fun newInstance(): SplashFragment {
            return SplashFragment()
        }
    }

    private var viewBinding: FragmentSplashBinding? = null

    private var disposable: Disposable? = null

    @Inject
    lateinit var preferencesTool: PreferenceTool

    override fun onAttach(context: Context) {
        super.onAttach(context)
        requireContext().appComponent.inject(this)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        viewBinding = FragmentSplashBinding.inflate(inflater)
        return viewBinding?.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        init()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        disposable?.dispose()
        viewBinding = null
    }

    private fun init() {
        try {
            val drawable =
                ContextCompat.getDrawable(requireContext(), lib.toolkit.base.R.drawable.image_onlyoffice_text)
            viewBinding?.appCompatImageView?.setImageDrawable(drawable)
        } catch (e: InflateException) {
            FirebaseUtils.addCrash("Inflate error when start")
            e.printStackTrace()
        } catch (e: Resources.NotFoundException) {
            FirebaseUtils.addCrash("Inflate error when start")
            e.printStackTrace()
        }
        disposable = Observable.just(1).delay(DELAY_SPLASH.toLong(), TimeUnit.MILLISECONDS)
            .subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe {
                if(!preferencesTool.isPasscodeLockEnable) {
                    MainActivity.show(requireContext())
                    requireActivity().finish()
                } else {
                    PasscodeActivity.show(requireContext())
                }
            }
    }
}